client/my-sites/plugins/marketplace/marketplace-test/index.tsx

```diff
@@ -5,58 +5,113 @@ import React, { useEffect, useState } from 'react';
+	const pluginDetails = useSelector( ( state ) => getPlugins( state, [ selectedSiteId ] ) );
+	const isRequestingForSite = useSelector( ( state ) =>
+		isRequestingForSites( state, [ selectedSiteId ] )
 	);
+	const yoastPluginOnSite = useSelector( ( state ) =>
+		getPluginOnSite( state, selectedSiteId, 'wordpress-seo-premium' )
+	);
+	const contactFormPlugin = useSelector( ( state ) =>
+		getPluginOnSite( state, selectedSiteId, 'contact-form-7' )
+	);
+
 	const [ infiniteLoopCount, setInfiniteLoopCount ] = useState( 0 );
+
+	const dispatch = useDispatch();
+	const transferDetails = useSelector( ( state ) => getAutomatedTransfer( state, selectedSiteId ) );
+	const eligibilityDetails = useSelector( ( state ) => getEligibility( state, selectedSiteId ) );
 	const marketplacePages = [
 		{ name: 'Plugin Details Page', path: '/marketplace/product/details/wordpress-seo' },
 		{ name: 'Loading Page', path: '/marketplace/product/setup' },
 		{ name: 'Domains Page', path: '/marketplace/domain' },
 		{ name: 'Thank You Page', path: '/marketplace/thank-you' },
 	];
-	const dispatch = useDispatch();
-	const transferDetails = useSelector( ( state ) =>
-		getAutomatedTransfer( state, selectedSite?.ID ?? 0 )
-	);
 
 	//Infinite Loop
 	useEffect( () => {
-		setTimeout( () => {
-			dispatch( fetchAutomatedTransferStatus( selectedSite?.ID ?? 0 ) );
-			dispatch( requestEligibility( selectedSite?.ID ?? 0 ) );
-			setInfiniteLoopCount( ( l ) => l + 1 );
-		}, 4000 );
-	}, [ infiniteLoopCount, fetchAutomatedTransferStatus, selectedSite ] );
+		if ( ! transferDetails.fetchingStatus ) {
+			setTimeout( () => {
+				dispatch( fetchAutomatedTransferStatus( selectedSite?.ID ?? 0 ) );
+				dispatch( requestEligibility( selectedSite?.ID ?? 0 ) );
+				setInfiniteLoopCount( ( l ) => l + 1 );
+			}, 5000 );
+		}
+	}, [ infiniteLoopCount, selectedSite, dispatch, transferDetails.fetchingStatus ] );
+
+	//Infinite Loop
+	useEffect( () => {
+		if ( selectedSiteId ) {
+			setTimeout( () => {
+				dispatch( fetchAutomatedTransferStatus( selectedSiteId ) );
+				dispatch( requestEligibility( selectedSiteId ) );
+				setInfiniteLoopCount( ( l ) => l + 1 );
+			}, 4000 );
+		}
+	}, [ infiniteLoopCount, selectedSite, dispatch, selectedSiteId ] );
 
 	const { ID, URL, domain, options = {} } = selectedSite;
 	const { is_wpcom_atomic, is_automated_transfer } = options;
+
+	const allBlockingMessages = getBlockingMessages( translate );
+	const holds = eligibilityDetails.eligibilityHolds || [];
+	const raisedBlockingMessages = holds
+		.filter( ( message: any ) => allBlockingMessages[ message ] )
+		.map( ( message: string ) => allBlockingMessages[ message ] );
+	const hardBlockSingleMessages = holds.filter(
+		( message: string ) => message !== 'TRANSFER_ALREADY_EXISTS' || ! allBlockingMessages[ message ]
+	);
+	const hasHardBlock =
+		isAtomicSiteWithoutBusinessPlan( holds ) || hardBlockSingleMessages.length > 0;
```

Comment:
Yes I was tempted to do some refactoring here because the plugins page actually uses most of this code. But however that would impact the plugins page and will require some additional effort right now. I will put in a comment indicating the need for a reactor here!

Reply:
I also agree with you on DRY! 👍